Initialization Fix

tbreimer14 opened the pull request:
This closes #145

CalebLItalien commented on frontend/src/lexer/token.rs:
What character does CFALSE represent? Is this C99 compliant?

tbreimer14 replied:
I figured we should have a token for a "false" primitive, in the same way we have CTRUE for a true primitive. I'm a bit confused - are we representing bools in a different way?

CalebLItalien replied:
Booleans aren't a part of C99 standard, they're a provided via a standard library. The ternary operator "?" or CTRUE is a different thing, just sounds similar

CalebLItalien commented on frontend/src/parser/parse_block.rs:
I think you're wrapping the variable name twice with identifier nodes.

tbreimer14 replied:
What exactly do you mean? Here's the structure it's creating for the tests:

AST: 
TopLevelExpression
	Initialization
		Variable)
			Identifier(y)
			Type(Boolean)
		AssignedValue
			Literal(true)

I just kept the same structure from the one test that was there before. Should the structure be different?

tbreimer14 replied:
What exactly do you mean? Here's the structure it's creating for the tests:

```
TopLevelExpression
	Initialization
		Variable)
			Identifier(y)
			Type(Boolean)
		AssignedValue
			Literal(true)
```


I just kept the same structure from the one test that was there before. Should the structure be different?

tbreimer14 replied:
Where exactly is this happening?

`parse_block.rs`

```
52: let identifer_node: ASTNode = ASTNode::new(SyntaxElement::Identifier(variable_name));
53: var_node.add_child(identifer_node);
```

On these lines, variable_name is a string, which gets added to a new Identifier Syntax Element, which is then packaged as a Node and added to the variable node. Am I missing something here?

CalebLItalien replied:
Oh you're right my bad I misread the PR initially. Disregard my comment

CalebLItalien commented on frontend/tests/parser_tests.rs:
Before closing this issue, make sure to uncomment out the rest of the tests in parser_tests and symbol_table_stack_tests. They were commented out because of this initialization problem.

CalebLItalien reviewed (commented):
Good work! A few comments to resolve before we merge this.

tbreimer14 commented:
Removed CFalse and fixed some tests in the last commit. There are still a lot still commented out though; since there is no longer a function token, all tests that create functions no longer work and this necessitates some rewriting of the parser. In my opinion, that's out of the scope of this issue and I was going to make a new issue for this, if that's alright.

tbreimer14 commented:
Note to anyone looking at this PR, I'm planning to tackle issue #168 in this PR as well, so don't merge just yet!

CalebLItalien reviewed (approved):
Excellent work!
